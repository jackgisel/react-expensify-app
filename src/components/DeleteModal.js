import React from 'react';
import Modal from 'react-modal';

const DeleteModal = (props) => (
    <Modal
        isOpen={props.showModal}
        onRequestClose={props.handleCancel}
        contentLabel="Confirm Deletion"
        closeTimeoutMS={200}
        className="modal"
        ariaHideApp={false}
    >
        <h4 className="modal__title">Are you sure you want to delete this item?</h4>
        <p className="modal__body">{props.expenseTitle} for {props.expenseAmount}</p>
        <div className="modal__actions">
            <button className="button" onClick={props.handleClearSelected}>Confirm Delete</button>
            <button className="button--secondary" onClick={props.handleCancel}>Cancel</button>
        </div>
    </Modal>
);

export default DeleteModal;