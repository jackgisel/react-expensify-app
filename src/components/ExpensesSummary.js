import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-list';

export const ExpensesSummary = ({ expensesCount, expensesTotal, totalExpenses }) => {
    const expensesWord = expensesCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">
                    Viewing <span>{expensesCount}</span> {expensesWord} totaling <span>{formattedExpensesTotal}</span>
                </h1>
                <p>There are {totalExpenses - expensesCount} expenses not showing due to the filters</p>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    );
};

const mapStateToProp = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses),
        totalExpenses: state.expenses.length
    };
};

export default connect(mapStateToProp)(ExpensesSummary);