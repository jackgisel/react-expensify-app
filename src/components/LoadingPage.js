import React from 'react';

const LoadingPage = () => (
    <div className="loader">
        <img className="loader__img" src="/images/loader.gif" alt="Loading Image"/>
    </div>
);

export default LoadingPage;