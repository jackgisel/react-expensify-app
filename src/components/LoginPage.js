import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startGoogle, startGithub }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify</h1>
            <p>It's time to get your expenses under control</p>
            <div className="box-layout__action">
                <button className="button" onClick={startGoogle}>Login with Google</button>
            </div>
            <div className="box-layout__action">
                <button className="button--secondary" onClick={startGithub}>Login with Github</button> 
            </div>    
        </div> 
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startGoogle: () => dispatch(startLogin("google")),
    startGithub: () => dispatch(startLogin("github"))
});

export default connect(undefined, mapDispatchToProps)(LoginPage);