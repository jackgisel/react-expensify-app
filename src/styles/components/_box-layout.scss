.box-layout {
    align-items: center;
    background: url('/images/bg.jpg');
    background-size: cover;
    display: flex;
    height: 100vh;
    justify-content: center;
    width: 100vw;
}

.box-layout__box {
    background: fade-out($color: white, $amount: .15);
    border-radius: 3px;
    padding: $l_size $m_size;
    text-align: center;
    width: 25rem;
}

.box-layout__title {
    margin: 0 0 $m_size 0;
    line-height: 1;
}

.box-layout__action {
    padding: $s_size;
}