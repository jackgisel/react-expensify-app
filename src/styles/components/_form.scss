.form {
    display: flex;
    flex-direction: column;
    > * {
        margin-bottom: $m_size;
    }
}

.form__error {
    margin: 0 0 $m_size 0;
    font-style: italic;
}