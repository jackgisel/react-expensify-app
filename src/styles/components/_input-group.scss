.input-group {
    display: flex;
    flex-direction: column;
    margin-bottom: $m_size;

    @media(min-width: $desktop-breakpoint){
        flex-direction: row;
        margin-bottom: $l_size;
    }
}

.input-group__item {
    margin-bottom: $s_size;

    @media(min-width: $desktop-breakpoint) {
        margin: 0 $s-size 0 0;
    }
}