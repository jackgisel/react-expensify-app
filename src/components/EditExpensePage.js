import React from 'react';
import { connect } from 'react-redux';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';
import DeleteModal from './DeleteModal';

export class EditExpensePage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showModal: false
    };
  }
  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };
  onRemove = (expense) => {
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  };
  onCancel = () => {
    this.setState({
      showModal: false
    });
  }
  onShowModal = () => {
    this.setState({
      showModal: true
    });
  }
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Editing Expense: <span>{this.props.expense.description}</span></h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm 
            expense={this.props.expense}
            onSubmit={this.onSubmit}
          />
          <button className="button--secondary" onClick={this.onShowModal}>Remove Expense</button>
          <DeleteModal 
            showModal={this.state.showModal} 
            handleClearSelected={this.onRemove}
            handleCancel={this.onCancel}
            expenseTitle={this.props.expense.description}
            expenseAmount={this.props.expense.amount} 
           />
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id),
  }
};

const mapDispatchToProps = (dispatch) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
