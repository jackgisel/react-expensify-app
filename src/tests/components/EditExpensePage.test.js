import React from 'react';
import { shallow } from 'enzyme';
import { EditExpensePage } from '../../components/EditExpensePage';
import expense from '../fixtures/expense';

let wrapper, editExpense, removeExpense, history;

beforeEach(() => {
    editExpense = jest.fn();
    removeExpense = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(<EditExpensePage 
        editExpense={editExpense} 
        removeExpense={removeExpense} 
        history={history}
        expense={expense}
        />);
});

test('should render EditExpensePage', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should handle editExpense', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expense);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(editExpense).toHaveBeenLastCalledWith(expense.id, expense);
});

test('should handle removeExpense', () => {
    wrapper.find('button').prop('onClick')(expense);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(removeExpense).toHaveBeenLastCalledWith({ id: expense.id });
});