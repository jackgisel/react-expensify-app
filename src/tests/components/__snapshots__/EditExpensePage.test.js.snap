// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpensePage 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 400,
        "createdAt": 0,
        "description": "in n out",
        "id": "4",
        "note": "moreeee food",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
